apps/redi-connect/src/pages/app/applications/useApplicationsFilter.ts

```diff
@@ -0,0 +1,35 @@
+import { RedMatch } from '@talent-connect/shared-types'
+
+interface Props {
+  applicants: RedMatch[]
+}
+
+export const useApplicationsFilter = ({ applicants }: Props) => {
+  const pendingApplications = applicants.filter(
+    (applicant) => applicant.status === 'applied'
+  )
+  const hasPendingApplications = Boolean(pendingApplications.length)
+
+  const hasAcceptedApplications = applicants.some(
+    (applicant) =>
+      applicant.status === 'accepted' || applicant.status === 'completed'
+  )
+
+  const hasDeclinedApplications = applicants.some(
+    (applicant) => applicant.status === 'declined-by-mentor'
+  )
+
+  const hasCancelledApplications = applicants.some(
+    (applicant) =>
+      applicant.status === 'cancelled' ||
+      applicant.status === 'invalidated-as-other-mentor-accepted'
+  )
+
+  return {
+    pendingApplications,
+    hasPendingApplications,
+    hasAcceptedApplications,
+    hasDeclinedApplications,
+    hasCancelledApplications,
+  }
```

Comment:
If we continue using the context, I think these can also be in the context. See context as a big state that holds everything related. These seem to be related to the applications filters so I would keep them together in the context. What do you think?